### PR TITLE
[NWO] Split AWS into supported/community collections

### PR DESCRIPTION
**Pull request by jillr**
AWS will be split into core/community collections, and additional modules promoted into core to eliminate core depending on community in CI.

### CODE REVIEW

**Comment by jillr** [comment]
Current proposal is to move to core:
```
aws_az_info
aws_caller_info
cloudformation_info
ec2_ami_info
ec2_ami
ec2_group_info
ec2_instance  # this is just an executive decision - we want people to stop using ec2.py and it’s in good shape
ec2_instance_info
ec2_snapshot_info
ec2_tag
ec2_vol_info
ec2_vpc_dhcp_option
ec2_vpc_dhcp_option_info
```
To remove testing dependencies `ansible.core` would have on `ansible.community`.  These modules should be reasonably supportable.

**Comment by s-hertel** [comment]
This is sort of a random question, but do you know what's happening to module_utils/cloud.py? (since AWS retry logic relies on it)

**Comment by jillr** [comment]
@s-hertel I'm going to copy/fork it

**Comment by jillr** [comment]
@s-hertel I can't actually request your review, but what do you think of the additional modules I'm proposing to put in core?  This eliminates the core collection depending on the community collection for tests.

**Comment by s-hertel** [comment]
@jillr Besides ec2_vpc_dhcp_option, which we discussed and have a plan for, I am fine with the list. Over the weekend I may write tests for ec2_vpc_dhcp_option to make it easier to port. It's nice most of them have decent test coverage. shipit

File: scenarios/nwo/community.yml [hunk 1]
```diff
@@ -78,6 +32,7 @@ amazon:
   - cloud/amazon/_elb_classic_lb_facts.py
   - cloud/amazon/_elb_target_facts.py
   - cloud/amazon/_elb_target_group_facts.py
+  - cloud/amazon/dynamodb_ttl.py
```
Comment (s-hertel) [inline scenarios/nwo/community.yml]: minor nit - this was fine below so it followed alphabetical order

Comment (jillr) [reply scenarios/nwo/community.yml]: Good catch thanks!

**Comment by jillr** [comment]
To update from the proposal on 2020-02-14, the promoting modules list is:
```
aws_az_info
aws_caller_info
cloudformation_info
ec2_ami_info
ec2_ami
ec2_elb_lb
ec2_eni 
ec2_eni_info
ec2_group_info
ec2_snapshot_info
ec2_tag
ec2_vol_info
ec2_vpc_dhcp_option
ec2_vpc_dhcp_option_info
```
`ec2_instance[_info]` brought in far too many additional dependencies, so while I'd like to promote this sooner than later to encourage use it's too soon to do it now.

**Pull request merged**